src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -368,4 +387,12 @@ FireStoreDirectoryEntry computeDirectory(Firestore firestore, String snapshotId,
         return dirEntry;
     }
 
+    private String getProjectIdForFSContainer(FSContainerInterface container) {
```

Comment:
This is unfortunate, but I can't think of any better way to do it.

Reply:
Agreed that this defeats the purpose of the interface in this case. I debated deleting the interface class completely, but there is still the getId() method there and anyway thought I'd check with the group first.

Reply:
I took a look and think the class is still useful.

I wonder if there are other places where a base class representing either a dataset or a snapshot would be useful. Food for thought...

Reply:
I think some common base class/interface for each of the below class types could be useful.
_Summary, _DataProject, _DataProjectSummary
Perhaps a good candidate for the next chunk of cleanup/restructuring.